src/ca/mcgill/ecse211/lab4/Lab4.java

```diff
@@ -0,0 +1,109 @@
+
+/**
+ * This class contains the main method which implements the hardware required to execute the fourth
+ * laboratory. The purpose of laboratory 4 is to localize the robot given an arbitrary starting
+ * position. In other words, it is placed at a starting position, and it has to gather data using
+ * the ultrasonic sensor and the light sensor to orient itself towards the origin of the grid
+ * system, i.e (0,0). A reference system is defined in terms of x,y and theta in order to localize
+ * the robot. The UltrasonicLocalizer class is first implemented to orient the robot at the required
+ * heading. Then, upon the user's prompt, the LightLocalizer class is implemented to drive the robot
+ * to the origin of the grid system.
+ * 
+ * @author1 Cristian Ciungu
+ * @author2 Hao Shu
+ * @version 12-02-2019
+ * 
+ */

```

Comment:
This Javadoc is well written, but it's in the wrong place! It should be immediately above this line:

```java
public class Lab4 {
```

To see if any of your Javadocs are correct, hover over what they refer to (in this case `Lab4`) with your mouse pointer.